controller/Booklet.php

```diff
@@ -55,49 +57,52 @@ class Booklet extends AbstractBookletController
      * @access public
      * @author CRP Henri Tudor - TAO Team - {@link http://www.tao.lu}
      * @return void
+     * @throws common_ext_ExtensionException
      */
-    public function index()
+    public function index(): void
     {
         $this->defaultData();
         $this->setView('index.tpl');
     }
 
     /**
      * Edit action
-     * @throws \tao_models_classes_MissingRequestParameterException
-     * @throws \tao_models_classes_dataBinding_GenerisFormDataBindingException
+     * @throws core_kernel_persistence_Exception
+     * @throws common_ext_ExtensionException
+     * @throws tao_models_classes_MissingRequestParameterException
+     * @throws tao_models_classes_dataBinding_GenerisFormDataBindingException
      */
-    public function editBooklet()
+    public function editBooklet(): void
     {
         $this->defaultData();
 
-        $clazz           = $this->getCurrentClass();
-        $instance        = $this->getCurrentInstance();
-        $myFormContainer = new EditForm($clazz, $instance);
+        $currentInstance = $this->getCurrentInstance();
 
-        $myForm = $myFormContainer->getForm();
+        $this->setData('isPreviewEnabled', $currentInstance->getLabel() !== 'in progress');
 
-        $fileResource = $this->getClassService()->getAttachment($instance);
         try {
-            $file = $this->getServiceLocator()->get(FileReferenceSerializer::SERVICE_ID)->unserialize($fileResource);
-            $allowDownload = ($file instanceof File);
-        } catch (\common_Exception $e) {
-            $allowDownload = false;
+            /** @var FileReferenceSerializer $fileReferenceSerializer */
+            $fileReferenceSerializer = $this->getServiceLocator()->get(FileReferenceSerializer::SERVICE_ID);
```

Comment:
Not sure if this line should be included into the `try` block.
I mean, if it fails here, then something is definitely very wrong with a set up, and it's better to fail fast and for good, instead of hiding the issue.

Reply:
Fixed